### PR TITLE
add header for zipcode

### PR DESCRIPTION
**Pull request by jfewii**

### CODE REVIEW

File: postal-injection/src/main/webapp/cluster_map.js [hunk 1]
```diff
@@ -45,21 +50,37 @@ async function createClusterMap() {
...
 }
 
+function closePreviousInfoWindow() {
+  if (previousInfoContent.length > 0) {
+    // detach the info-window from the marker
+    previousInfoContent[0].set('marker', null);
+    /* and close it */
+    previousInfoContent[0].close();
+    /* blank the array */
+    previousInfoContent.length = 0;
```
Comment (andreahu12) [inline postal-injection/src/main/webapp/cluster_map.js:60]: nit: be consistent about comment format. you've been using // elsewhere so maybe stick to that?

**Comment by jennygaoo** [comment]
add a screenshot?

**Comment by jfewii** [comment]
image

File: postal-injection/src/main/webapp/cluster_map.js [hunk 2]
```diff
@@ -45,21 +50,37 @@ async function createClusterMap() {
...
+  let zipCode = clusterInfo[0].zipCode;
   let latitude = clusterInfo[0].zipCodeLatitude;
   let longitude = clusterInfo[0].zipCodeLongitude;
+  let infoWindowHeader = '';
   let infoWindowContent = '';
 
   let firstCoordinates = new google.maps.LatLng(latitude, longitude);
   allCoordinates.push(firstCoordinates);
   for (let i = 0; i < clusterInfo.length; i++) {
+    const currentZipCode = clusterInfo[i].zipCode;
```
Comment (andreahu12) [inline postal-injection/src/main/webapp/cluster_map.js:77]: createMarkerInfo is rather long. In general methods should not be longer than 21 lines. Try breaking the logic up into helper methods to help with readability. Particularly I think the for-loop can be broken up into hasSameCoordinatesAsPreviousCluster(), getInfoWindowContent()

File: postal-injection/src/main/webapp/cluster_map.js [hunk 3]
```diff
@@ -45,21 +50,37 @@ async function createClusterMap() {
...
     const currentLatitude = clusterInfo[i].zipCodeLatitude;
     const currentLongitude = clusterInfo[i].zipCodeLongitude;
+
     // Check to see if coordinates are the same as the previous ones
```
Comment (jennygaoo) [inline postal-injection/src/main/webapp/cluster_map.js:81]: supernit: change comment to explain _why_ you need to check to see if the coordinates & previous coordinates are the same

File: postal-injection/src/main/webapp/cluster_map.js [hunk 4]
```diff
@@ -82,42 +115,20 @@ function createMarkerInfo(clusterInfo) {
...
-//     minPalletSize,
-//     threshold,
-//     postOfficeToCustomerRate,
-//   ];
-// }
-
-async function fetchClusterData(
-    zipCode, regularRate, palletRate, minPalletSize, threshold,
-    postOfficeToCustomerRate) {
+async function fetchClusterData(getUserInputObject) {
```
Comment (jennygaoo) [inline postal-injection/src/main/webapp/cluster_map.js:124]: nit: change getUserInputObject to userInputObject b/c you're not actually getting userInputObject

File: postal-injection/src/main/webapp/cluster_map.js [hunk 5]
```diff
@@ -71,8 +92,20 @@ function createMarkerInfo(clusterInfo) {
...
+
+    // To ensure the zip code header is only added once
+    if (infoWindowHeader != currentInfoWindowHeader) {
+      infoWindowContent += OPEN_INFO_HEAD_DIV + zipCode + CLOSE_INFO_HEAD_DIV +
+          OPEN_INFO_WINDOW_DIV + shippingMethod + OPTIMAL_SHIPPING_METHOD_HTML +
+          warehouseId + CLOSE_INFO_WINDOW_DIV;
+      infoWindowHeader = currentInfoWindowHeader;
+    } else {
+      infoWindowContent += OPEN_INFO_WINDOW_DIV + shippingMethod +
+          OPTIMAL_SHIPPING_METHOD_HTML + warehouseId + CLOSE_INFO_WINDOW_DIV;
```
Comment (jgao23) [inline postal-injection/src/main/webapp/cluster_map.js:106]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: postal-injection/src/main/webapp/cluster_map.js (hunk 1 above)
Comment (andreahu12) [reply postal-injection/src/main/webapp/cluster_map.js:60]: comments still inconsistent

File: postal-injection/src/main/webapp/cluster_map.js [hunk 6]
```diff
@@ -45,21 +50,34 @@ async function createClusterMap() {
       new MarkerClusterer(clusterMap, markers, {imagePath: 'cluster_images/m'});
 }
 
+function closePreviousInfoWindow() {
+  if (previousInfoContent.length > 0) {
+    previousInfoContent[0].set('marker', null);
+    previousInfoContent[0].close();
+    previousInfoContent.length = 0;
+  }
```
Comment (andreahu12) [inline postal-injection/src/main/webapp/cluster_map.js:58]: I didn't mean to remove the comments. Like I said in my previous comment

/**/ or //, but not both

**Comment by andreahu12** [comment]
I think Joany and my comments are still unresolved. If they cannot be addressed, just explain why and resolve the comment. Or that you plan to do it in a followup.

**Comment by jfewii** [comment]
> I think Joany and my comments are still unresolved. If they cannot be addressed, just explain why and resolve the comment. Or that you plan to do it in a followup.

I plan on following up on those comments

**Pull request merged**